### PR TITLE
`azurerm_shared_image` & `azurerm_shared_image_version` - support for specialized images by setting generalized to false

### PR DESCRIPTION
**Pull request by ArcturusZhang**
This PR adds support for the specialized shared image, with the following changes:

* `azurerm_shared_image` - add `os_state` attribute to indicate this image definition is generalized or specialized.
* `azurerm_shared_image_version` - add `os_disk_snapshot_id` attribute to support create new image version using snapshot, which also enables you create new version for a specialized image.

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Tests are passing: 
```
=== RUN   TestAccAzureRMSharedImageVersion_basic
=== PAUSE TestAccAzureRMSharedImageVersion_basic
=== CONT  TestAccAzureRMSharedImageVersion_basic
--- PASS: TestAccAzureRMSharedImageVersion_basic (2421.21s)
=== RUN   TestAccAzureRMSharedImageVersion_storageAccountTypeLrs
=== PAUSE TestAccAzureRMSharedImageVersion_storageAccountTypeLrs
=== CONT  TestAccAzureRMSharedImageVersion_storageAccountTypeLrs
--- PASS: TestAccAzureRMSharedImageVersion_storageAccountTypeLrs (1294.24s)
=== RUN   TestAccAzureRMSharedImageVersion_storageAccountTypeZrs
=== PAUSE TestAccAzureRMSharedImageVersion_storageAccountTypeZrs
=== CONT  TestAccAzureRMSharedImageVersion_storageAccountTypeZrs
--- PASS: TestAccAzureRMSharedImageVersion_storageAccountTypeZrs (1212.54s)
=== RUN   TestAccAzureRMSharedImageVersion_specializedImageVersion
=== PAUSE TestAccAzureRMSharedImageVersion_specializedImageVersion
=== CONT  TestAccAzureRMSharedImageVersion_specializedImageVersion
--- PASS: TestAccAzureRMSharedImageVersion_specializedImageVersion (1087.86s)
=== RUN   TestAccAzureRMSharedImageVersion_requiresImport
=== PAUSE TestAccAzureRMSharedImageVersion_requiresImport
=== CONT  TestAccAzureRMSharedImageVersion_requiresImport
--- PASS: TestAccAzureRMSharedImageVersion_requiresImport (1509.59s)
PASS
```

**Comment by ArcturusZhang** [comment]
Hi @katbyte I have synced with master... and now the tests seem passing
image
Interesting that first time one test fails, and second time I only run that failed test, and it passes.

File: website/docs/r/shared_image.html.markdown [hunk 1]
```diff
@@ -74,6 +74,8 @@ The following arguments are supported:
 
 * `release_note_uri` - (Optional) The URI containing the Release Notes associated with this Shared Image.
 
+* `os_state` - (Optional) The state of Operating System present in this Shared Image. Possible values are `Generalized` and `Specialized`. Defaults to `Generalized`. Changing this forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/shared_image.html.markdown:77]: After some internal discussion could we rename this to `generalized = bool` ? this is at heart a boolean true/false and os_state really doesn't convey what its doing.

Comment (ArcturusZhang) [reply website/docs/r/shared_image.html.markdown:77]: Great idea! I will take care of this.

**Comment by ArcturusZhang** [comment]
Hi @katbyte I have resolved your comments, please have a look, thanks

File: website/docs/r/shared_image.html.markdown [hunk 2]
```diff
@@ -60,14 +60,16 @@ The following arguments are supported:
...
-* `os_type` - (Required) The type of Operating System present in this Shared Image. Possible values are `Linux` and `Windows`.
+* `os_type` - (Required) The type of Operating System present in this Shared Image. Possible values are `Linux` and `Windows`. Changing this forces a new resource to be created.
 
 ---
 
 * `description` - (Optional) A description of this Shared Image.
 
 * `eula` - (Optional) The End User Licence Agreement for the Shared Image.
 
+* `generalized` - (Optional) Should the Operating System present in this Shared Image be generalized? Defaults to `true`. Changing this forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/shared_image.html.markdown:71]:
```suggestion
* `generalized` - (Optional) Should the Operating System present in this Shared Image be generalized? Defaults to `true`. Setting this to `false` creates a specialized imagine which is the equivalent of sysprepping in windows. Changing this forces a new resource to be created.
```

File: website/docs/d/shared_image.html.markdown [hunk 3]
```diff
@@ -43,6 +43,8 @@ The following attributes are exported:
 
 * `location` - The supported Azure location where the Shared Image Gallery exists.
 
+* `generalized` - Is the Operating System present in this Shared Image generalized or not.
```
Comment (katbyte) [inline website/docs/d/shared_image.html.markdown:46]:
```suggestion
* `generalized` - Is the Operating System present in this Shared Image generalized or not. `false` indicates this image is specialized which for windows images means they have been sysprepped.
```

**Pull request merged**